src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -278,4 +314,14 @@ private function addServiceToAmbiguousType($id, $type)
         }
         $this->ambiguousServiceTypes[$type][] = $id;
     }
+
+    static private function getResourceMetadataForMethod(\ReflectionMethod $method)
+    {
+        $methodArgumentsMetadata = array();
+        foreach ($method->getParameters() as $parameter) {
+            $methodArgumentsMetadata[] = (string) $parameter;
```

Comment:
and casting to string makes it much easier to deal with typehints on non-existent classes on PHP 5.x (as PHP 5 does not have ``getType`` and ``getClass`` breaks in this case)